File: ProjectEuler_CSharp/Fibonacci.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace ProjectEuler_CSharp
{
    public static class Fibonacci
    {
        public static IEnumerable<long> UpTo(long maxValue)
        {
            long i = 1, j = 2;
            while (i <= maxValue) {
                yield return i;
                (i, j) = (j, i + j);
            }
        }

    }
}

File: ProjectEuler_FSharp/Fibonacci.fs
module ProjectEuler_FSharp.Fibonacci

let UpTo maxValue =
    let rec recFibonnacci a b =
        let c = a + b
        if c <= maxValue then c :: (recFibonnacci b c)
        else []
    1 :: 2 :: (recFibonnacci 1 2)